podman/tests/integration/utils.py

```diff
@@ -0,0 +1,98 @@
+#   Copyright 2020 Red Hat, Inc.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+#
+"""Integration Test Utils"""
+
+import os
+import shutil
+import subprocess
+import tempfile
+import time
+from typing import List, Optional
+
+import podman.tests.errors as errors
+
+
+class PodmanLauncher:
+    """Podman service launcher"""
+
+    def __init__(
+        self,
+        socket_uri: str,
+        podman_path: Optional[str] = None,
+        timeout: int = 0,
+        privileged: bool = False,
+        debug: bool = False,
+    ) -> None:
+        """create a launcher and build podman command"""
+        podman_exe: str = podman_path
+        if not podman_exe:
+            podman_exe = shutil.which('podman')
+        if podman_exe is None:
+            raise errors.PodmanNotInstalled()
+
+        self.socket_file: str = socket_uri.replace('unix://', '')
+        self.proc = None
+
+        self.cmd: List[str] = []
+        if privileged:
+            self.cmd.append('sudo')
+
+        self.cmd.append(podman_exe)
+
+        if debug:
+            self.cmd.append("--log-level=debug")
+
+        self.cmd.extend(
+            [
+                "system",
+                "service",
+                f"--time={timeout}",
+                socket_uri,
+            ]
+        )
+
+        self.log_file = tempfile.NamedTemporaryFile(
+            prefix="podman_integration_", suffix=".log", delete=False
+        )
+        self.log_file.write(f"Podman command {' '.join(self.cmd)}\n".encode())
+        self.log_file.flush()
```

Comment:
This really should use the logging module rather than creating a file like this.

Reply:
I will before duplicating, currently this is a one off.

Reply:
@mwhahaha I went with the file because capturing stderr/stdout via logging conflicted with unittest capturing stdout/stderr. I'll keep playing with it for a future PR.

Reply:
K. I think there are ways around that and we might just need to include a test friendly logging configuration